CMakeLists.txt

```diff
@@ -100,6 +100,14 @@ if(MSYS
     set(UNIX_LIKE TRUE)
 endif()
 
+# Set the build output paths
+if (CMAKE_PROJECT_NAME STREQUAL PROJECT_NAME)
```

Comment:
https://github.com/GMLC-TDC/HELICS/blob/develop/interfaces/python/CMakeLists.txt#L13-L15

Reply:
I suppose either would work, both have possible incorrect results, this has the advantage of no additional variable, the other is a little more obvious what it is doing.   I think I switch to this to move away from additional variables but could move back.

Reply:
PARENT_DIRECTORY should only be set if it has been added as a subdirectory of the build. What are situations where it would have incorrect results?

Reply:
well if CMakeLIst.txt were included in a top level directory via include() or add_subdirectory were called without defining a project.    Not totally sure those situations are allowed by CMake or not.

Reply:
It looks like not defining a project is okay, since it uses the path of the CMakeLists.txt/source directory that added it as a subdirectory.

include() does break it... using include for an external dependency seems wrong (and a CMakeLists.txt file), but with the number of results include(*CMakeLists.txt) gives...

Reply:
I think in the case of add_subdirectory with no parent project we still would want HELICS to act as if it were the parent project and set everything accordingly.

Reply:
In nearly all cases either would work.  This is what was recommended by some folks working on the CLI11 library and some folks who have given tutorials on CMake, so is what I have been using for a while.